server/artemis/model.py

```diff
@@ -0,0 +1,53 @@
+# This file handles the actual processing of messages,
+# and sends them back to the parent process.
+# @author: mia-pi-git
+from detoxify import Detoxify;
+import sys;
+import time;
+import json;
+
+model_name = 'unbiased'
+debug = False
+
+if "multilingual" in sys.argv:
+	model_name = 'multilingual'
+elif "small" in sys.argv:
+	model_name += '-small'
+
+if "debug" in sys.argv:
+	debug = True
+
+model = Detoxify(model_name)
+
+def now():
+	return int(time.time() * 1000)
+
+logfile = open("logs/artemis.log", "a")
+def log(message):
+	if not debug:
+		return None
+
+	logfile.write(str(now()) + ":" + str.rstrip(message) + "\n")
+	logfile.flush()
+
+print("ready", flush=True)
+log("ready")
+try:
+	for line in sys.stdin:
+		log("in:" + line)
+		parts = line.split("|")
+		out = parts.pop(0) + "|"
+		try:
+			res = model.predict("|".join(parts))
+			for key in res: res[key] = str(res[key]) # json.dumps doesn't like floats
+			out += json.dumps(res)
+		except BaseException as e:
+			out += json.dumps({'error': f"{e}"})
```

Comment:
If you're using f-strings here (thus preventing the script from running on Python version that don't support f-strings), why not use them for the rest of the string interpolation you do?

Reply:
Yeah, probably should. Done in https://github.com/smogon/pokemon-showdown/pull/8662/commits/ae3f15d7577854f769423d08b4c5220a64cf4b74.